collectors/ebpf.plugin/ebpf.c

```diff
@@ -633,6 +638,557 @@ static inline int parse_disable_apps(char *ptr)
+    link_hostnames(&network_viewer_opt.included_hostnames, value);
+
+    value = appconfig_get(&collector_config, EBPF_NETWORK_VIEWER_SECTION, "excluded hostnames", NULL);
+    link_hostnames(&network_viewer_opt.excluded_hostnames, value);
+
+    value = appconfig_get(&collector_config, EBPF_NETWORK_VIEWER_SECTION, "included ips", NULL);
+    parse_values((void **)&network_viewer_opt.included_ips, value, isascii, parse_ip_list);
+
+    value = appconfig_get(&collector_config, EBPF_NETWORK_VIEWER_SECTION, "excluded ips", NULL);
+    parse_values((void **)&network_viewer_opt.excluded_ips, value, isascii, parse_ip_list);
+}
+
+/**
+ * Link dimension name
+ *
+ * Link user specified names inside a link list.
+ *
+ * @param port the port number associated to the dimension name.
+ * @param hash the calculated hash for the dimension name.
+ * @param name the dimension name.
+ */
+static void link_dimension_name(char *port, uint32_t hash, char *value)
+{
+    int test = str2i(port);
+    if (test < NETDATA_MINIMUM_PORT_VALUE || test > NETDATA_MAXIMUM_PORT_VALUE){
+        error("The dimension given (%s = %s) has an invalid value and it will be ignored.", port, value);
+        return;
+    }
+
+    ebpf_network_viewer_dim_name_t *w;
+    w = callocz(1, sizeof(ebpf_network_viewer_dim_name_t));
+
+    w->name = strdupz(value);
+    w->hash = hash;
+
+    w->port = (uint16_t) test;
+
+    ebpf_network_viewer_dim_name_t *names = network_viewer_opt.names;
+    if (unlikely(!names)) {
+        network_viewer_opt.names = w;
+    } else {
+        for (; names->next; names = names->next) {
+            if (names->port == w->port) {
+                error("Dupplicated definition for the service '%u', the name %s will be ignored. ", hash, names->name);
+                freez(names->name);
+                names->name = w->name;
+                names->hash = w->hash;
+                freez(w);
+                return;
+            }
+        }
+        names->next = w;
+    }
+
+#ifdef NETDATA_INTERNAL_CHECKS
+    info("Adding values %s( %u) to dimension name list used on network viewer",w->name, w->port);
+#endif
+}
+
+/**
+ * Parse service Name section.
+ *
+ * This function gets the values that will be used to overwrite dimensions.
+ */
+static void parse_service_name_section()
+{
+    struct section *co = appconfig_get_section(&collector_config, EBPF_SERVICE_NAME_SECTION);
+    if (co) {
+        struct config_option *cv;
+        for (cv = co->values; cv ; cv = cv->next) {
+            link_dimension_name(cv->name, cv->hash, cv->value);
+        }
+    }
+
+    //Always associated the default port to Netdata
+    ebpf_network_viewer_dim_name_t *names = network_viewer_opt.names;
+    uint16_t default_port = htons(19999);
+    for (; names->next; names = names->next) {
+        if (names->port == default_port) {
+            return;
+        }
+    }
+
+    link_dimension_name("19999", simple_hash("19999"), "Netdata");
```

Comment:
thanks, now it is correct, however question still remains if hardcoding netdata to port 19999 is good idea. What if user runs netdata on another port?

Reply:
This is a good point, if the user runs in another port, the section `[service name]`needs being configured. For we read different ports, we would need to either read `netdata.conf` to get different values for `default port` and `bind to` or Netdata would need to pass these values as argument when the collector is called.

Reply:
> For we read different ports, we would need to either read netdata.conf to get different values for default port and bind to or Netdata would need to pass these values as argument when the collector is called.

or we could expose env variable like `NETDATA_LISTEN_PORT` and use it here

Reply:
@ilyam8 I liked your suggestion, but this affects not only this plugin, but all plugins, so I think for we move forward with this, it would need another PR, because I will change the next function

https://github.com/netdata/netdata/blob/664d180dc7a5e34334a4f94399e5663eb5c0be8a/daemon/main.c#L672

Reply:
> but all plugins

At the momenot it affects nothing, if you need this for this plugin it looks reasonable to add it in this PR. 

A PR with only new env var looks strange without context.

@mfundul @stelfrag @underhood what do u think guys?

Reply:
I think I agree with you @ilyam8 I worry for the confusion of users who use netdata on different ports

Reply:
All right, I created the environment variable to set the dimension when it is necessary.